ntbs-service/Services/UserSearchService.cs

```diff
@@ -2,48 +2,60 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Castle.Core.Internal;
+using MoreLinq;
 using ntbs_service.DataAccess;
 using ntbs_service.Models;
 using ntbs_service.Models.Entities;
 
 namespace ntbs_service.Services
 {
-    public interface ICaseManagerSearchService
+    public interface IUserSearchService
     {
-        Task<(IList<User> caseManagers, int count)> OrderAndPaginateQueryableAsync(
+        Task<(IList<User> users, int count)> OrderAndPaginateQueryableAsync(
             string searchKeyword,
             PaginationParametersBase paginationParameters);
     }
 
-    public class CaseManagerSearchService : ICaseManagerSearchService
+    public class UserSearchService : IUserSearchService
     {
         private readonly IReferenceDataRepository _referenceDataRepository;
+        private readonly IUserRepository _userRepository;
 
-        public CaseManagerSearchService(IReferenceDataRepository referenceDataRepository)
+        public UserSearchService(IReferenceDataRepository referenceDataRepository, IUserRepository userRepository)
         {
             _referenceDataRepository = referenceDataRepository;
+            _userRepository = userRepository;
         }
 
-        public async Task<(IList<User> caseManagers, int count)> OrderAndPaginateQueryableAsync(
+        public async Task<(IList<User> users, int count)> OrderAndPaginateQueryableAsync(
             string searchKeyword,
             PaginationParametersBase paginationParameters)
         {
             var searchKeywords = searchKeyword.Split(" ")
                 .Where(x => !x.IsNullOrEmpty())
                 .Select(s => s.ToLower()).ToList();
-            var caseManagers = await _referenceDataRepository.GetAllCaseManagersOrdered();
+            
+            var allPhecs = await _referenceDataRepository.GetAllPhecs();
+            var filteredPhecs = allPhecs.Where(phec => searchKeywords.Any(s => phec.Name.ToLower().Contains(s)));
+
+            var caseManagersAndRegionalUsers = (await _referenceDataRepository.GetAllCaseManagersOrdered())
```

Comment:
I don't think there's any point calling both repos here, as the results from the reference data repo are a subset of those from the user repo.

Reply:
On a similar note, I'm a bit concerned about the performance of pulling all our users into memory, but I think we need to adopt a "wait and see if it's good enough" approach

Reply:
The issue with this is that LINQ can't translate a portion the `searchKeywords.Any(...)` so I couldn't leave this as a queryable